src/source/Ice/Network.c

```diff
@@ -11,15 +11,77 @@ STATUS getLocalhostIpAddresses(PKvsIpAddress destIpList, PUINT32 pDestIpListLen,
     UINT32 ipCount = 0, destIpListLen;
     BOOL filterSet = TRUE;
 
+#ifdef _WIN32
+    DWORD rv, size;
+    PIP_ADAPTER_ADDRESSES adapter_addresses, aa = NULL;
+    PIP_ADAPTER_UNICAST_ADDRESS ua;
+#else
     struct ifaddrs *ifaddr = NULL, *ifa = NULL;
+#endif
     struct sockaddr_in *pIpv4Addr = NULL;
     struct sockaddr_in6 *pIpv6Addr = NULL;
 
     CHK(destIpList != NULL && pDestIpListLen != NULL, STATUS_NULL_ARG);
     CHK(*pDestIpListLen != 0, STATUS_INVALID_ARG);
-    CHK(getifaddrs(&ifaddr) != -1, STATUS_GET_LOCAL_IP_ADDRESSES_FAILED);
 
     destIpListLen = *pDestIpListLen;
+#ifdef _WIN32
+    rv = GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, NULL, NULL, &size);
+    CHK(rv == ERROR_BUFFER_OVERFLOW, STATUS_GET_LOCAL_IP_ADDRESSES_FAILED);
+
+    adapter_addresses = (PIP_ADAPTER_ADDRESSES)malloc(size);
+
+    rv = GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, NULL, adapter_addresses, &size);
+    CHK(rv == ERROR_SUCCESS, STATUS_GET_LOCAL_IP_ADDRESSES_FAILED);
+    
+    for (aa = adapter_addresses; aa != NULL && ipCount < destIpListLen; aa = aa->Next) {
+
+        char ifa_name[BUFSIZ];
+        memset(ifa_name, 0, BUFSIZ);
+        WideCharToMultiByte(CP_ACP, 0, aa->FriendlyName, wcslen(aa->FriendlyName), ifa_name, BUFSIZ, NULL, NULL);
+
+        for (ua = aa->FirstUnicastAddress; ua != NULL; ua = ua->Next) {
+
+            if (filter != NULL) {
+                DLOGI("Callback set to allow network interface filtering");
+                // The callback evaluates to a FALSE if the application is interested in black listing an interface
+                if (filter(customData, ifa_name) == FALSE) {
+                    filterSet = FALSE;
+                } else {
+                    filterSet = TRUE;
+                }
+            }
+
+            // If filter is set, ensure the details are collected for the interface
+            if (filterSet == TRUE) {
+                int family = ua->Address.lpSockaddr->sa_family;
+
+                if (family == AF_INET) {
+                    destIpList[ipCount].family = KVS_IP_FAMILY_TYPE_IPV4;
+                    destIpList[ipCount].port = 0;
+
+                    pIpv4Addr = (struct sockaddr_in*)(ua->Address.lpSockaddr);
+                    MEMCPY(destIpList[ipCount].address, &pIpv4Addr->sin_addr, IPV4_ADDRESS_LENGTH);
+                } else {
+                    destIpList[ipCount].family = KVS_IP_FAMILY_TYPE_IPV6;
+                    destIpList[ipCount].port = 0;
+
+                    pIpv6Addr = (struct sockaddr_in6*)(ua->Address.lpSockaddr);
+                    // Ignore link local: not very useful and will add work unnecessarily
+                    // Ignore site local: https://tools.ietf.org/html/rfc8445#section-5.1.1.1
+                    if (IN6_IS_ADDR_LINKLOCAL(&pIpv6Addr->sin6_addr) || IN6_IS_ADDR_SITELOCAL(&pIpv6Addr->sin6_addr)) {
+                        continue;
```

Comment:
Re-structure the loop to avoid control-flow keywords, especially continue; these are dangerous to use in more complex scenarios. For example, here you could check if (!blah) { MEMCPY}